xapian-letor/bin/trainletor.cc

```diff
@@ -0,0 +1,146 @@
+/** @file trainletor.cc
+ * @brief Command line tool to prepare Letor training file and train the model
+ */
+/* Copyright (C) 2004,2005,2006,2007,2008,2009,2010,2015 Olly Betts
+ * Copyright (C) 2011 Parth Gupta
+ * Copyright (C) 2016 Ayush Tomar
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include <xapian.h>
+#include <xapian-letor.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <map>
+#include <set>
+#include <math.h>
```

Comment:
`<cmath>` is preferred, unless there's a good reason for needing `<math.h>`, like something platform extension not in ISO C++11.  Also `<cstdio>` rather than `<stdio.h>` above, and it's unlikely you want both `<cstdlib>` and `<stdlib.h>`.

Also sorting this list would make it easier for future coders.